ws/src/server_builder.rs

```diff
@@ -37,6 +38,28 @@ impl From<io::Error> for Error {
 	}
 }
 
+impl fmt::Display for Error {
+    fn fmt(&self, f: &mut fmt::Formatter) -> fmt::Result {
+        match *self {
+            Error::Io(ref e) => e.fmt(f),
+            Error::WebSocket(ref e) => e.fmt(f),
+        }
+    }
+}
+
+impl ::std::error::Error for Error {
+    fn description(&self) -> &str {
+        "Starting the JSON-RPC WebSocket server failed"
```

Comment:
I have been looking in the rustc code for any standard on how the `description` and `Display` are implemented. But I could not find a common denominator for enums containing sub-errors.